docs/sources/operations/storage/logs-deletion.md

```diff
@@ -21,6 +21,8 @@ Log entry deletion relies on configuration of the custom logs retention workflow
 
 Enable log entry deletion by setting `retention_enabled` to true in the compactor's configuration and setting and `deletion_mode` to `filter-only` or `filter-and-delete` in the runtime config.
 
+> **Warning:** Please be very careful when enabling retention, it's strongly recommended to enable versioning on your objects in object storage to allow for recovery from accidental misconfiguration of a retention setting. If you wish to enable delete capability but do not want to enforce any retention, make sure you configure retention_period with a value of `0s`. 
```

Comment:
>I was debating if we should be more explicit here that previous versions of Loki did not properly honor the zero value. This would be somewhat out of character of having our versioned docs mention previous versions but I wonder if that's prudent here.

I would definitely call out this change of behavior in the Release Notes.  And it might be worth a note in the Configuration topic to flag the change.